### PR TITLE
Reconstruction

### LINKED ISSUE #5: Reconstruction Module
Given our bisecting strategy (#4) we want to be able to reconstruct the split outputs. Create a module the combines all the split netcdf outputs into a single output.

### PR DESCRIPTION
**Pull request by cairosanders**
Resolves #5 
Uses [`mf_dataset`](http://xarray.pydata.org/en/stable/generated/xarray.open_mfdataset.html?highlight=mf#xarray.open_mfdataset) to combine split files into one dataset or [`open_dataset`](http://xarray.pydata.org/en/stable/generated/xarray.open_dataset.html?highlight=open) if no split occurs 

Test locally for online tests:
- tests/test_reconstruct.py::test_reconstruct_dataset_online
- tests/test_process.py::test_process
- tests/test_request_opendap.py::test_request_opendap

### CODE REVIEW

File: orca/reconstruct.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    temp_paths = [temp.name for temp in temp_files]
+    dataset = (
+        open_mfdataset(temp_paths, combine="nested", concat_dim="time")
+        if len(temp_files) > 1
+        else open_dataset(temp_paths[0])
+    )
+    (temp.close for temp in temp_files)
+    dataset.to_netcdf(outfile)
+
+    return outfile
```
Comment (nikola-rados) [inline orca/reconstruct.py:14]: One thing we should consider here is that the `to_netcdf` method is pretty slow. I think it is actually the main bottleneck of the process and something I am learning more about in issue #13. Looking briefly at the docs you linked, I don't think we can use `urls` in place of `temp_paths` for `open_mfdataset`. If it is possible we may want to reconstruct directly after splitting to save some time.

File: orca/reconstruct.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from xarray import open_mfdataset, open_dataset
+
+
+def reconstruct_dataset(temp_files, outfile):
+    temp_paths = [temp.name for temp in temp_files]
+    dataset = (
+        open_mfdataset(temp_paths, combine="nested", concat_dim="time")
```
Comment (nikola-rados) [inline orca/reconstruct.py:7]: Is it not possible to call `open_mfdataset` with a single path?

File: tests/conftest.py [hunk 3]
```diff
@@ -57,3 +60,21 @@ def test_session(meta_session):
     meta_session.commit()
 
     yield meta_session
+
+
+@pytest.fixture
+def temp_files():
+    file_paths = [
+        f"{str(Path(__file__).parents[0])}/data/tasmin_sClimSD_anusplin_historical_19710101-20001231_time_0.nc",
+        f"{str(Path(__file__).parents[0])}/data/tasmin_sClimSD_anusplin_historical_19710101-20001231_time_1.nc",
```
Comment (nikola-rados) [inline tests/conftest.py:69]: Could we not use `resource_filename` here?

**Review by nikola-rados — CHANGES_REQUESTED** [review]
This looks pretty good, just a few things to fix up!

File: orca/reconstruct.py (hunk 2 above)
Comment (cairosanders) [reply orca/reconstruct.py:7]: Good point. You probably can, I will try it.

File: requirements.txt [hunk 4]
```diff
@@ -2,4 +2,6 @@ click==7.1.2
 modelmeta==0.2.0
 psycopg2==2.8.6
 SQLAlchemy==1.3.23
-xarray==0.16.2
\ No newline at end of file
+xarray==0.16.2
+toolz==0.11.1
+dask==2021.2.0
```
Comment (nikola-rados) [inline requirements.txt:7]: If `toolz` is coming from `dask` I think you can use something like `dask[complete]` to ensure all the dependencies are installed.

File: orca/process.py [hunk 5]
```diff
@@ -4,20 +4,30 @@
...
 def process_request(
-    connection_string, unique_id, thredds_base, variable, lat, lon, log_level="INFO"
+    connection_string,
+    unique_id,
+    thredds_base,
+    variable,
+    lat,
+    lon,
+    outfile,
+    log_level="INFO",
```
Comment (nikola-rados) [inline orca/process.py:14]: Ensure these are available as cli options.

**Review by nikola-rados — APPROVED** [review]
Looks good just make sure to update the script options.

**Pull request merged**